addons/wowl/static/src/components/dropdown/dropdown_item.ts

```diff
@@ -0,0 +1,44 @@
+import { Component } from "@odoo/owl";
+
+export enum ParentClosingMode {
+  None = "none",
+  ClosestParent = "closest",
+  AllParents = "all",
+}
+
+export interface DropdownItemClickedPayload {
+  itemPayload: Object;
+  parentClosingMode: ParentClosingMode;
+}
+
+export class DropdownItem extends Component {
+  static template = "wowl.DropdownItem";
+  static props = {
+    onClickPayload: {
+      type: Object,
+      optional: true,
+    },
+    parentClosingMode: {
+      type: ParentClosingMode,
+      optional: true,
+    },
+  };
+  static defaultProps = {
+    onClickPayload: null,
+    parentClosingMode: ParentClosingMode.AllParents,
+  };
+
+  /**
+   * Handlers
+   */
+  fire(ev: MouseEvent) {
+    const payload: DropdownItemClickedPayload = {
+      itemPayload: this.props.onClickPayload,
+      parentClosingMode: this.props.parentClosingMode,
+    };
+    this.trigger("dropdown-item-clicked", payload);
+
+    // Stopping propagation as we replace the event with a custom one
+    ev.stopPropagation();
```

Comment:
you can do it directly in xml: `t-on-click.stop="fire"`

Reply:
That's what I did first.
But after some thoughts I wanted to be more explicit on that.
I generally do not like to stop the propagation of common DOM events and I wanted to add a comment on why in that case it makes sense.

Reply:
ok